hydroflow_cli_integration/src/lib.rs

```diff
@@ -268,27 +267,49 @@ pub struct ConnectedDemux<T: Connected> {
     sink: Option<DynSink<(u32, T::Input)>>,
 }
 
-// a copy of `Drain` that uses `io::Error` instead of `Infallible`
-struct IoErrorDrain<T> {
+#[pin_project]
+struct DemuxDrain<T> {
     marker: PhantomData<T>,
+    #[pin]
+    sinks: HashMap<u32, DynSink<T>>,
```

Comment:
Seems reasonable, some concern about `dyn Sink` perf if this is on the hot path, would need to check via benchmarking